src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooBuildPlanService.java

```diff
@@ -546,4 +565,57 @@ private DockerConfiguration dockerConfigurationFor(ProgrammingLanguage programmi
+     * Creates a custom Build Plan for a Programming Exercise using Aeolus. If the build plan could not be created, null is
+     * returned. Otherwise, the key of the created build plan is returned. Not the whole key is returned, only the build plan
+     * key, without the project key. Example: "PROJECT-BUILDPLANKEY" is created -> "BUILDPLANKEY" is returned, same as the
+     * default build plan creation.
+     *
+     * @param programmingExercise   the programming exercise for which to create the build plan
+     * @param buildPlanId           the id of the build plan
+     * @param planDescription       the description of the build plan
+     * @param repositoryUrl         the url of the assignment repository
+     * @param testRepositoryUrl     the url of the test repository
+     * @param solutionRepositoryUrl the url of the solution repository
+     * @param auxiliaryRepositories List of auxiliary repositories to be included in the build plan
+     * @return the key of the created build plan, or null if it could not be created
+     */
+    private String createCustomAeolusBuildPlanForExercise(ProgrammingExercise programmingExercise, String buildPlanId, String planDescription, VcsRepositoryUrl repositoryUrl,
+            VcsRepositoryUrl testRepositoryUrl, VcsRepositoryUrl solutionRepositoryUrl, List<AuxiliaryRepository.AuxRepoNameWithUrl> auxiliaryRepositories)
+            throws ContinuousIntegrationBuildPlanException {
+        if (aeolusBuildPlanService.isEmpty()) {
+            return null;
+        }
+        if (programmingExercise.getBuildPlanConfiguration() == null) {
+            // If no custom build plan configuration is present, we will use the default build plan creation
+            return null;
+        }
+        String assignedKey = null;
+        try {
+            Windfile windfile = programmingExercise.getWindfile();
+            Map<String, AeolusRepository> repositories = aeolusBuildPlanService.get().createRepositoryMapForWindfile(programmingExercise.getProgrammingLanguage(),
+                    programmingExercise.getBranch(), programmingExercise.getCheckoutSolutionRepository(), repositoryUrl, testRepositoryUrl, solutionRepositoryUrl,
+                    auxiliaryRepositories);
+
+            String resultHookUrl = artemisServerUrl + NEW_RESULT_RESOURCE_API_PATH;
+            windfile.setPreProcessingMetadata(buildPlanId, programmingExercise.getProjectName(), this.gitUser, resultHookUrl, planDescription, repositories);
+            String generatedKey = aeolusBuildPlanService.get().publishBuildPlan(windfile, AeolusTarget.BAMBOO);
+            /*
+             * Aeolus returns the key of the build plan in the format "PROJECT-BUILDPLANKEY". To stay consistent with the
+             * default build plan creation, we return only the build plan key.
+             */
+            if (generatedKey != null && generatedKey.contains("-")) {
+                assignedKey = generatedKey.split("-")[1];
+            }
```

Comment:
I am not too familiar with how the buildplan key is generated, wouldn't this break if I choose a buildplankey that uses '-' (in case that char is allowed?)

e.g. 
BUIDLPLANKEY = not-split-properly
-> PROJECT-BUILDPLANKEY = project-not-split-properly

If I am not mistaken this would result in the `assignedKey = not` but should be `not-split-properly`

I think that it would be a better approach to take everything after the first hyphen

Reply:
if you don't mind I would revisit this in a follow up PR as we kinda depend on this PR for other features.